src/features/productDetails/ProductDetails.tsx

```diff
@@ -0,0 +1,46 @@
+import {formattedPrice} from "../../utils/productUtils.ts";
+import {Link, useParams} from "react-router-dom";
+import {useQuery} from "@tanstack/react-query";
+import {getProductWithId} from "../../services/api.ts";
+
+const ProductDetails = () => {
+    const {productId} = useParams()
+    const productQuery = useQuery({queryKey: [productId], queryFn: () => getProductWithId(productId ?? '')})
+
+
```

Comment:
is it necessary to make another query for the product details?

Reply:
I'm actually not sure if it's completely necessary, this was my solution to navigate with the router to the product details page. If you think there is a better way I would greatly appreciate any suggestion 😄

Reply:
all good, was just a genuine thought. I would have used selected state to get the selected product id and display its details. really not sure about the best way, just a thought 😃

Reply:
Yeah, definitely should have a discussion about it next time